ViAn/GUI/mainwindow.cpp

```diff
@@ -568,76 +559,58 @@ void MainWindow::prepare_menu(const QPoint & pos) {
+                input_switch_case(ACTION::ADD_VIDEO, dir);
+            }
+        }else{
+            set_status_bar("No project selected.");
         }
     } else {
-        set_status_bar("No project selected.");
+        set_status_bar("Multiple or no projects selected.");
     }
 }
 
 /**
  * @brief MainWindow::play_video
  *  Loads selected video, flips playbutton to pause
  *  plays video from beginning
- *
  */
 void MainWindow::play_video() {
+    MyQTreeWidgetItem *my_project;
+    my_project = (MyQTreeWidgetItem*)ui->ProjectTree->selectedItems().first();
+    mvideo_player->load_video(my_project->name.toStdString());
     //Used for rescaling the source image for video playback
     emit resize_video_frame(ui->videoFrame->width(),ui->videoFrame->height());
-
     enable_video_buttons();
-    mvideo_player->load_video(selectedVideo->name.toStdString());
     iconOnButtonHandler->set_icon("pause", ui->playPauseButton);
     video_slider->setMaximum(mvideo_player->get_num_frames());
 }
 
 /**
- * @brief MainWindow::set_selected_project
- * puts an arrow pointing at the selected project
- * @param newSelectedProject
- */
-void MainWindow::set_selected_project(MyQTreeWidgetItem *newSelectedProject){
-    set_selected(selectedProject, newSelectedProject);
-}
-
-/**
- * @brief MainWindow::set_selected_video
- * puts an arrow pointing at the selected video
- * @param newSelectedVideo
- */
-void MainWindow::set_selected_video(MyQTreeWidgetItem *newSelectedVideo) {
-    set_selected(selectedVideo, newSelectedVideo);
-    set_selected_project((MyQTreeWidgetItem*)selectedVideo->parent());
-}
-
-/**
- * @brief MainWindow::set_selected
- * @param selected old
- * @param new_selected
- */
-void MainWindow::set_selected(MyQTreeWidgetItem *&selected, MyQTreeWidgetItem *new_selected) {
-    if(selected == nullptr) {
-        selected = new_selected;
-    } else if (selected != new_selected) {
-        selected = new_selected;
-    }
-}
-
-/**
  * @brief MainWindow::on_actionSave_triggered
  * saves project which is selected in tree view,
  * checks if there is one
  */
 void MainWindow::on_actionSave_triggered() {
-    if(selectedProject != nullptr) {
-        this->fileHandler->save_project(this->selectedProject->id);
-        std::string text = "Saved project " + this->selectedProject->name.toStdString();
+    QTreeWidgetItem *project;
+    MyQTreeWidgetItem *my_project;
+    if(ui->ProjectTree->selectedItems().size() == 1) {
+        project = ui->ProjectTree->selectedItems().first();
+        if (!project->parent()){
+            my_project = (MyQTreeWidgetItem*)project;
+        }else if (project->parent()){
+            my_project = (MyQTreeWidgetItem*)project->parent();
+        }
+        this->fileHandler->save_project(my_project->id);
+        std::string text = "Saved project " + my_project->name.toStdString();
         set_status_bar(text);
-    } else {
+    }else {
```

Comment:
Either force user to select a project or make sure that it dosent matter what file is selected.